lib/puppet/functions/libkv/deletetree.rb

```diff
@@ -1,86 +1,117 @@
-# vim: set expandtab ts=2 sw=2:
+# Deletes a whole folder from the configured backend.
+#
+# @author https://github.com/simp/pupmod-simp-libkv/graphs/contributors
 #
-# @author Dylan Cochran <dylan.cochran@onyxpoint.com>
 Puppet::Functions.create_function(:'libkv::deletetree') do
```

Comment:
Could this be `libkv::tree::delete`?

The more I work through this, the more moving each action into an OO-style layout makes sense to me. It'll keep from having weird name explosion as time moves on.

Reply:
Matches the API from https://github.com/docker/libkv#interface.  I kind of like the simplicity.  Perhaps others should weigh in.